#!/usr/bin/env Rscript

###### 
# This script will output a shell script that will merge paired-end reads for JA16414. 
# ./join_paired_ends_JA16414.R 
# input: 	list of files to merge
# output: 	shell script to merge reads
######

###### PARAMETERS ##########
# Set the parameters:
today <- Sys.Date()											# Set the date that will go on the end of the files generated by this script
today <- format(today, format="%m%d%y")
#############################



##### List files in directory:
bFiles <- list.files("../data/raw/sequencing_data/JA16414/", pattern="*_I[1]_001.fastq.gz$")

# Get list of sequencingIDs:
sequencingID <- sapply(bFiles, function(x) {strsplit(x, split="_L001")[[1]][1]})

# shortcuts for script:
inpath <- "../data/raw/sequencing_data/JA16414/"
fend <- "_L001_R1_001.fastq.gz"
rend <- "_L001_R2_001.fastq.gz"
iend <- "_L001_I1_001.fastq.gz"
outpath <- "../data/processed/QIIME/1_join_paired_ends/"



##### Make script:
script <- c("#!/bin/bash/")
script <- c(script, "source qiime_config_1.9")
for (i in 1:length(bFiles)) {
	script <- c(script, paste0("join_paired_ends.py -f ", inpath, sequencingID[i], fend, " -r ", inpath, sequencingID[i], rend, " -b ", inpath, sequencingID[i], iend, " -o ", outpath, " -j 200"))
}



##### Write out script:
write.table(script, "join_paired_ends_JA16414.sh", sep="\n", quote=FALSE, col.names=FALSE, row.names=FALSE)

print("DONE!")